nautobot_firewall_models/views/nat_policy.py

```diff
@@ -61,6 +61,31 @@ class NATPolicyUIViewSet(NautobotUIViewSet):
 
     lookup_field = "pk"
 
+    def get_extra_context(self, request, instance=None):
```

Comment:
Just a note that in "recent" versions of 2.4.x and later, you can alternately provide `custom_view_base_action` and `custom_view_additional_permissions` parameters to the `@action` decorator as an alternative approach to overriding `get_extra_context` and `get_required_permission`.

For example, `devices` below could be:

```py
@action(
    detail=True,
    methods=["post"],
    custom_view_base_action="change",  # i.e. change_natpolicy
    custom_view_additional_permissions=["dcim.change_device"],
)
```